shuup/front/basket/objects.py

```diff
@@ -85,12 +86,26 @@ def can_change_quantity(self):
 
 
 class BaseBasket(OrderSource):
-    def __init__(self, request, basket_name="basket"):
+    def __init__(self, request, key=None):
         super(BaseBasket, self).__init__(request.shop)
-        self.basket_name = basket_name
         self.request = request
-        if request:
-            self.ip_address = request.META.get("REMOTE_ADDR")
+        self.ip_address = request.META.get("REMOTE_ADDR")
+
+        # FIXME Somehow, not all requests provide a session attribute during
```

Comment:
It does not sounds good to store this at the session because IMO it does not change the fact that the user still have to put then `sessionid=xxxxxx` in the header of each request. we must avoid that, specially for mobile devices. the end platform will have to take care of the JWT token and also the `sessionid` (does not makes sense for me). basket should be a item like Product and you can get access to it through the url `/product/{ID}/`.